codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCodegenVisitor.kt

```diff
@@ -372,6 +376,24 @@ open class ServerCodegenVisitor(
         }
     }
 
+    override fun longShape(shape: LongShape) {
+        if (shape.isDirectlyConstrained(codegenContext.symbolProvider)) {
+            logger.info("[rust-server-codegen] Generating a constrained long $shape")
+            rustCrate.withModule(ModelsModule) {
+                ConstrainedLongGenerator(codegenContext, this, shape).render()
+            }
+        }
+    }
```

Comment:
Out of ignorance: Don't we need an `else { super(shape) }` branch? Same for `byteShape`.

Reply:
I don't think so? 
The default handler is a no-op for all types.